### DISCUSSION TITLE
Algorithmic 3D modeling using Pythonocc

### DISCUSSION THREAD

**Discussion by Tanneguydv** [leon-thomm/Ryven#75] [Show and tell]
Hi,
I've been looking for a way to work on Pythonocc with a node editor, Ryven is a really good candidate for it and get an open Algorithmic 3D modeling python software.

To get it functional you will have to execute Ryven in an environment where pythonocc is installed.

Here a small example of a definition : [https://www.youtube.com/watch?v=lUNYstrfvmg](https://www.youtube.com/watch?v=lUNYstrfvmg)
![exemple_tutogithub](https://user-images.githubusercontent.com/81742654/131475698-b52b5ee8-55f6-4a1c-9d38-d6cfbfeb5d3f.jpg)
![exemple_tutogithub_1](https://user-images.githubusercontent.com/81742654/131475707-8ab43ba6-0d9d-425e-af25-2446ea230317.jpg)

It's just a beginning to explore the possibilities given by matching the two, I've just coded simple functions to see how it works and how it should be to perform complex operations.
But I'm quite new to Python and would need help to adapt Ryven to the specific needs of such a use case, I'm open to contributions and comments too!
Here the corresponding repo : [https://github.com/Tanneguydv/Pythonocc-nodes-for-Ryven](https://github.com/Tanneguydv/Pythonocc-nodes-for-Ryven)

**Comment by leon-thomm** [comment]
nice, this is amazing!

**Reply by Tanneguydv** [reply]
It's all thanks to your work!
Nice job with Ryven, I'm new to Python but I have plenty of ideas to continue working with pythonocc&Ryven
